### PR TITLE
feat: Add currentPeers RPC Endpoint

### PR DESCRIPTION
**Pull request by zachterrell57**
## Motivation

This endpoint was added to get an approximate list of all peers on the network

## Change Summary

- Added new Proto definitions for endpoint
- New accessor method for current peers
- New RPC endpoint for current peers `/v1/peerList`

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [x] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.
- [x] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

## Additional Context

Response looks like the following:
```
{
    "contacts": [
        {
            "gossipAddress": {
                "address": "84.247.175.196",
                "family": 4,
                "port": 2282,
                "dnsName": ""
            },
            "rpcAddress": {
                "address": "84.247.175.196",
                "family": 4,
                "port": 2283,
                "dnsName": ""
            },
            "excludedHashes": [],
            "count": 10694067,
            "hubVersion": "2023.12.27",
            "network": "FARCASTER_NETWORK_MAINNET",
            "appVersion": "1.9.2",
            "timestamp": 1705796040744
        },
      ...
```

---

## PR-Codex overview
This PR adds a new `currentPeers` API endpoint and related changes to support fetching current peer contacts in the Hub network.

### Detailed summary
- Added `currentPeers` RPC endpoint
- Implemented API endpoint for getting a list of current sync peers
- Updated documentation for the new API endpoint
- Added protobuf message for `ContactInfoResponse`
- Updated HTTP server to handle `currentPeers` API call
- Updated TypeScript files to include `ContactInfoResponse` and related functions

> The following files were skipped due to too many changes: `packages/core/src/protobufs/generated/request_response.ts`, `packages/hub-nodejs/src/generated/rpc.ts`, `packages/hub-web/src/generated/rpc.ts`, `packages/hub-web/src/generated/gossip.ts`, `packages/hub-nodejs/src/generated/gossip.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/hubble/src/network/sync/syncEngine.ts [hunk 1]
```diff
@@ -402,6 +402,10 @@ class SyncEngine extends TypedEmitter<SyncEvents> {
     return this.currentHubPeerContacts.get(peerId);
   }
 
+  public getCurrentHubPeerContacts() {
+    return this.currentHubPeerContacts;
```
Comment (sanjayprabhu) [inline apps/hubble/src/network/sync/syncEngine.ts:406]: We should just return the `.values()` so it's not possible to modify this outside of this class.

File: apps/hubble/src/rpc/httpServer.ts [hunk 2]
```diff
@@ -223,6 +224,13 @@ export class HttpAPIServer {
       this.grpcImpl.getInfo(call, handleResponse(reply, HubInfoResponse));
     });
 
+    //================peerList================
+    // @doc-tag: /peerList
+    this.app.get("/v1/peerList", (request, reply) => {
+      const call = getCallObject("getCurrentPeers", {}, request);
```
Comment (sanjayprabhu) [inline apps/hubble/src/rpc/httpServer.ts:230]: Let's keep the name consistent between the http and the rpc apis. Maybe rename the httpEndpoint to `currentPeers`?

**Review by sanjayprabhu — COMMENTED** [review]
Thanks! Just needs a few minor changes.

Also looks like build is failing because the docs need to be updated.

**Review by sds — APPROVED** [review]
Nice, thank you!

**Pull request merged**